tutor/commands/upgrade/common.py

```diff
@@ -66,3 +66,55 @@ def get_mongo_upgrade_parameters(
     mv '/openedx/data/ora2/SET-ME-PLEASE (ex. bucket-name)' /openedx/data/ora2/openedxuploads
 fi
 """
+
+def get_mysql_change_charset_query(
+    database: str,
+    charset: str,
+    collation: str,
+    query_to_append: str,
+) -> None:
+    """
+    Helper function to generate the mysql query to upgrade the charset and collation of tables
+    
+    Utilized in the `tutor local do change-charset-collation` command
+    """
+    return f"""
+            DROP PROCEDURE IF EXISTS UpdateTables;
+            DELIMITER $$
+
+            CREATE PROCEDURE UpdateTables()
+            BEGIN
+
+                DECLARE done INT DEFAULT FALSE;
+                DECLARE _table_name VARCHAR(255);
+                DECLARE cur CURSOR FOR
+                        SELECT table_name FROM information_schema.tables
+                        WHERE table_schema = '{database}' AND table_type = 'BASE TABLE' {query_to_append};
+                DECLARE CONTINUE HANDLER FOR NOT FOUND SET done = TRUE;
+                Select "The following tables will have their charset and collation updated:";
+                OPEN cur;
+                tables_loop: LOOP
+                    FETCH cur INTO _table_name;
+
+                    IF done THEN
+                    LEAVE tables_loop;
+                    END IF;
+
+                    SET FOREIGN_KEY_CHECKS = 0;
```

Comment:
Does this cause any issue when trying to do a partial migration that includes the `course_overviews_courseoverview` table? That's the only table I'm aware of that has a varchar primary key and foreign keys from other tables.

Reply:
I tested this out by upgrading only the `course_overviews_courseoverviewtab` table which has a foreign key to `course_overviews_courseoverview` to utf8mb4 and keeping the rest at utf8mb3. 

The LMS and CMS seemed to be working fine for me. I tested out the following things:

- Create new Course
- Publish changes in new course
- Enroll in new course and verify everything is working fine
- Edit an existing course
- Publish new course and verify changes can be seen by students on LMS